### PR TITLE
Optimization: only override request.user if user.is_anonymous

### PR DESCRIPTION
**Pull request by phlax**
This sets request.profile to request.user to maintain compatibility

### CODE REVIEW

**Comment by ta2-1** [comment]
@phlax can you recall me why we'll get rid of `request.profile` and use `request.user` instead?

**Comment by phlax** [comment]
@ta2-1 overriding the request.user is bad from a security pov, but also performance

using obscure vars ie request.profile is just another thing to learn/code/test - when django takes care of this already

what django also does is optimize anon views by *avoiding* a db call

unfortunately this PR doesnt take care of that - but i think we should asap

File: pootle/core/views.py [hunk 1]
```diff
@@ -74,9 +74,11 @@ def set_permissions(f):
     @functools.wraps(f)
     def method_wrapper(self, request, *args, **kwargs):
         User = get_user_model()
-        request.profile = User.get(self.request.user)
+        if request.user.is_anonymous():
+            request.user = User.get(request.user)
```
Comment (unho) [inline pootle/core/views.py]: Perhaps we should be updating `self.request.user` as well.

Comment (phlax) [inline pootle/core/views.py]: its the same object

Comment (unho) [inline pootle/core/views.py]: I should check before asking.

Comment (phlax) [inline pootle/core/views.py]: also this allows the deco to be used if self.request has not been set

**Comment by unho** [comment]
Ideally we should be getting rid of `request.profile` (see #3459).

lgtm

**Comment by phlax** [comment]
> Ideally we should be getting rid of request.profile (see #3459).

agreed.

File: pootle/core/views.py [hunk 2]
```diff
@@ -74,9 +74,11 @@ def set_permissions(f):
     @functools.wraps(f)
     def method_wrapper(self, request, *args, **kwargs):
         User = get_user_model()
-        request.profile = User.get(self.request.user)
```
Comment (ta2-1) [inline pootle/core/views.py]: @julen, as I understand we made extra query here. Now we make this query for anonymous only. This is optimisation, is this?

Comment (ta2-1) [inline pootle/core/views.py]: hmm... I checked now and I understand your comment
> user.is_anonymous() makes no DB calls and user.is_authenticated()
so I'm not sure now if this change was necessary. I'll check this in `pdb`.

Comment (julen) [inline pootle/core/views.py]: From my inspection and understanding it's not — I raised #4508 addressing it.

**Pull request merged**